Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/InstanceDomGenerator.cpp

```diff
@@ -200,38 +202,57 @@ namespace AzToolsFramework
             }
         }
 
-        void InstanceDomGenerator::UpdateContainerEntityInDomFromRoot(PrefabDom& instanceDom, const Instance& instance) const
+        void InstanceDomGenerator::UpdateContainerEntityInDomFromHighestAncestor(PrefabDom& instanceDom, const Instance& instance) const
         {
             // TODO: Modifies the function so it updates the transform only.
 
             InstanceClimbUpResult climbUpResult = PrefabInstanceUtils::ClimbUpToTargetOrRootInstance(instance, nullptr);
-            const Instance* rootInstancePtr = climbUpResult.m_reachedInstance;
-            AZStd::string relativePathFromRoot = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(climbUpResult.m_climbedInstances);
 
             // No need to update the instance DOM if the given instance is the root instance.
-            if (rootInstancePtr == &instance)
+            if (climbUpResult.m_reachedInstance == &instance)
             {
                 return;
             }
 
-            // Creates a path from the root instance to container entity of the given insance.
-            const AZStd::string pathToContainerEntity =
-                AZStd::string::format("%s/%s", relativePathFromRoot.c_str(), PrefabDomUtils::ContainerEntityName);
-            PrefabDomPath domPathToContainerEntity(pathToContainerEntity.c_str());
-
-            // Retrieves the container entity DOM of the root template DOM.
+            // Get the instance's container entity DOM from the top-most ancestor that has it.
+            // An ancestor DOM may not have the data if it has deleted the instance as an override
             PrefabDom containerEntityDom;
+
+            AZStd::vector<const Instance*>& climbedInstances = climbUpResult.m_climbedInstances;
+            const Instance* topInstance = climbUpResult.m_reachedInstance;
+            for (auto instanceIter = climbedInstances.crbegin(); instanceIter != climbedInstances.crend(); ++instanceIter)
             {
+                // Create a path from the current top instance to the container entity of the given instance.
+                AZStd::string relativePathFromTopInstance =
+                    PrefabInstanceUtils::GetRelativePathBetweenInstances(*topInstance, instance);
+                const AZStd::string pathToContainerEntity =
+                    AZStd::string::format("%s/%s", relativePathFromTopInstance.c_str(), PrefabDomUtils::ContainerEntityName);
+                PrefabDomPath domPathToContainerEntity(pathToContainerEntity.c_str());
+                const PrefabDom& topInstanceTemplateDom =
+                    m_prefabSystemComponentInterface->FindTemplateDom((*topInstance).GetTemplateId());
```

Comment:
nit: Looks like we can use AZ::DOM::Path to dynamically update the path, so we don't need to call `PrefabInstanceUtils::GetRelativePathBetweenInstances` for multiple times. I'm okay with the current version.

Reply:
I looked into using AZ::DOM::Path but it doesn't look like we'd gain much benefit at this time as it uses an AZStd::vector internally. How were you thinking it would be used?

Reply:
```
Level                   <-- topInstance
| SuperCar
  | Car
    | Wheel
      | TireEntity
```
My thought was, in the above example, we iterate from Level to Wheel. The path would change in each iteration:
1. /Instances/Instance_[SuperCar]/Insances/Instance_[Wheel]/Entities/Entity_[TireEntity]
2. /Instances/Instance_[Wheel]/Entities/Entity_[TireEntity]
3. /Entities/Entity_[TireEntity]

In each iteration, the path can be reconstructed by:
```cpp
// AZ::Dom::Path
path = AZ::Dom::Path(path.begin() + 2, path.end());  // remove the leading 2 entries
```